### PR TITLE
Vitals sampling customization

### LINKED ISSUE #1465: Make vitals more configurable
Make more aspects of vitals be more configurable:
- The time span covered by the short term table
- The time span covered by the long term table
- The sample interval for the long term table

### PR DESCRIPTION
**Pull request by schmelter-sap**
This change makes the size and sampling frequency of the vitals tables configurable.

fixes #1465

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Comment by schmelter-sap** [comment]
retest this please

File: src/hotspot/share/runtime/globals.hpp [hunk 1]
```diff
@@ -511,6 +511,19 @@ const int ObjectAlignmentInBytes = 8;
                                                                             \
   product(uintx, VitalsSampleInterval, 10,                                  \
           "Vitals sample rate interval in seconds (default 10)")            \
+          range(1, 24 * 3600)                                               \
+                                                                            \
+  product(uintx, VitalsShortTermTableHours, 1,                              \
+          "The size of the short term vitals table in hours")               \
+          range(1, 365 * 24)                                                \
+                                                                            \
+  product(uintx, VitalsLongTernSampleIntervalMinutes, 60,                   \
```
Comment (RealCLanger) [inline src/hotspot/share/runtime/globals.hpp:520]: Spelling: VitalsLongTernSampleIntervalMinutes -> VitalsLongTermSampleIntervalMinutes

File: src/hotspot/share/runtime/globals.hpp [hunk 2]
```diff
@@ -511,6 +511,19 @@ const int ObjectAlignmentInBytes = 8;
...
+          "Vitals sample interval for the long term table in minutes")      \
```
Comment (RealCLanger) [inline src/hotspot/share/runtime/globals.hpp:521]: Please align this description with the one for VitalsSampleInterval

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Review by ansteiner — APPROVED** [review]
LGTM

**Pull request merged**